src/main/scala/no/ndla/learningpathapi/service/UpdateServiceComponent.scala

```diff
@@ -133,13 +133,27 @@ trait UpdateServiceComponent {
             searchIndexService.indexDocument(updatedLearningPath)
           } else if (existing.isPublished) {
             searchIndexService.deleteDocument(updatedLearningPath)
+            deleteIsBasedOnReference(learningPathId, updatedLearningPath)
           }
 
           Some(converterService.asApiLearningpath(updatedLearningPath, Option(owner)))
         }
       }
     }
 
+    private def deleteIsBasedOnReference(learningPathId: Long, updatedLearningPath: domain.LearningPath) = {
+      if (updatedLearningPath.isPrivate || updatedLearningPath.isDeleted) {
```

Comment:
Denne sjekken er vel redundant? Du sjekker det utenfor, og her inne har du allerede bestemt deg for å slette alle som er basert på denne.